src/count-islands/index.js

```diff
@@ -1,3 +1,47 @@
-function countIslands(grid) {}
+function countIslands(grid) {
+  let queue = [];
+  let count = 0;
+  let item
+    while (grid.length === 1) {
+    grid = grid[0];
+  }
+  for (let i = 0; i < grid.length; i += 1) {
+    if (Array.isArray(grid[i])) {
+      for (let j = 0; j < grid[i].length; j += 1) {
+        if (grid[i][j] === 1) {
+          count += 1;
+          queue.push([i, j]);
+          while (queue.length !== 0) {
```

Comment:
put this while loop in a helper function and then call that function, that way the countIsLand() function looks cleaner